internal/http/services/appprovider/appprovider.go

```diff
@@ -106,34 +116,127 @@ func (s *svc) Handler() http.Handler {
 	})
 }
 
-func (s *svc) handleList(w http.ResponseWriter, r *http.Request) {
+func (s *svc) handleNew(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
+
 	client, err := pool.GetGatewayServiceClient(s.conf.GatewaySvc)
 	if err != nil {
 		ocmd.WriteError(w, r, ocmd.APIErrorServerError, "error getting grpc gateway client", err)
 		return
 	}
 
-	listRes, err := client.ListSupportedMimeTypes(ctx, &appregistry.ListSupportedMimeTypesRequest{})
+	if r.URL.Query().Get("template") != "" {
+		// TODO in the future we want to create a file out of the given template
+		ocmd.WriteError(w, r, ocmd.APIErrorInvalidParameter, "Template not implemented",
+			errtypes.NotSupported("Templates are not yet supported"))
+		return
+	}
+
+	target := r.URL.Query().Get("filename")
+	if target == "" {
+		ocmd.WriteError(w, r, ocmd.APIErrorInvalidParameter, "Missing filename",
+			errtypes.UserRequired("Missing filename"))
+		return
+	}
+	// stat the container
+	_, ocmderr, err := statRef(ctx, provider.Reference{Path: r.URL.Query().Get("container")}, client)
```

Comment:
I'm not sure about separate `filename` and `container` queries. The parent name can be extracted from the full path

Do we even want to perform this check? Because we allow `PUT`ting at paths where parents don't exist